samples/dnn/object_detection.cpp

```diff
@@ -104,44 +137,135 @@ int main(int argc, char** argv)
     else
         cap.open(parser.get<int>("device"));
 
-    // Process frames.
-    Mat frame, blob;
-    while (waitKey(1) < 0)
-    {
-        cap >> frame;
-        if (frame.empty())
+    bool process = true;
+
+    // Frames capturing thread
+    std::mutex framesQueueMutex;
```

Comment:
Mutex should go into `QueueFPS` class.

Reply:
Moved all the locks into the `QueueFPS`.